src/server/routes/oauth.ts

```diff
@@ -174,6 +235,73 @@ const authenticationHandler = async (
 			return redirectForGenericError(req, res);
 		}
 
+		const { user_groups, email_validated, sub, email } =
+			tokenSet.claims() as CustomClaims;
+
+		// We want to ensure that the user has a legacyIdentityId set in their
+		// Okta profile before carrying on. This is surfaced via the legacy_identity_id
+		// claim in the access token.
+		const refreshToken = tokenSet.refresh_token;
+		const { legacy_identity_id } = tokenSet.claims();
+		if (!refreshToken && !legacy_identity_id) {
```

Comment:
why doesn't this trigger the an error response similar to when the refresh token is there but we cannot fetch a valid token set ?

Reply:
Yeah, I think you're right - this is such a weird edge case that really no user should end up here! So we'll error here.